packages/annotations/src/add.ts

```diff
@@ -43,14 +44,19 @@ export interface IAddAnnotationsOptions extends IAddFeaturesOptions {
 export function addAnnotations(
   requestOptions: IAddAnnotationsOptions
 ): Promise<{
-  addResults?: IEditFeatureResult[];
+  addResults?: Array<IEditFeatureErrorResponse | IEditFeatureResult>;
 }> {
   /* istanbul ignore else */
   if (requestOptions.features && requestOptions.features.length) {
     requestOptions.features.forEach(anno => enrichAnnotation(anno));
   }
 
-  return addFeatures(requestOptions);
+  return addFeatures(requestOptions).then(response => {
+    return {
+      ...response,
+      addResults: checkResults(response.addResults)
+    };
+  });
```

Comment:
Does this align with the return type?
```
Promise<{
  addResults?: Array<IEditFeatureErrorResponse | IEditFeatureResult>;
}>
```

Reply:
yep, in fact TS complained when I made a typo and it wasn't aligned.

Reply:
even though it mentions nothing of the destructured response object. strange.

Reply:
never mind. i didn't see the destructuring at first.

Reply:
the destructuring doesn't seem necessary since your always overriding, but it's already merged.

Reply:
I did the destructuring b/c thought there was additional properties in the response. looking at the [API docs](https://developers.arcgis.com/rest/services-reference/add-features.htm) there may not be.